### PR TITLE
Fix the issue that window is not defined in @aws-amplify/ui in NodeJS

### LINKED ISSUE #3486: Getting the error of Window is not defined, after new installation of the node_modules with "youtube-iframe" package
**Describe the bug**
After installation of the @aws-amplify with the additional package of youtube-iframe. Getting the error of window is not defined.

**Screenshots**
Please check the following screenshot of the error. The app is working with previous node_modules without youtube-iframe package inside.

***You can turn on the debug mode to provide more info for us by setting window.LOG_LEVEL = 'DEBUG'; in your app.***

### LINKED ISSUE #3506: Amplify is totally crashed because of the Window is not defined error in youtube-iframe package
@powerful23, @robbevan ,@mlabieniec, @jordanranz, @mbahar, @richardzcode
After installation of the @aws-amplify with the additional package of youtube-iframe.

Getting the error of window is not defined in

1. node_modules/youtube-iframe/index.js 50:3
2. node_modules/@aws-amplify/ui/dist/aws-amplify-ui.js:1

**Can you please provide the previous working version without facing the issue of window is not defined in the following folders.**

### PR DESCRIPTION
**Pull request by powerful23**
*Issue #, if available:*
Fix the issue that `window` object exists in the bundle file which causes issues in Node env.
Fixes #3506 
Fixes #3486 
Related to #3615 

Related issue:
https://github.com/webpack/webpack/issues/6522
*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-ui/webpack.config.js [hunk 1]
```diff
@@ -11,6 +11,7 @@ module.exports = {
     library: "aws_amplify_ui",
     libraryTarget: "umd",
     umdNamedDefine: true,
+    globalObject: '(typeof self !== "undefined" ? self : this)',
```
Comment (sammartinez) [inline packages/amplify-ui/webpack.config.js]: So there is a callout to just set it to `this` in the [webpack documentation](https://webpack.js.org/configuration/output/#outputglobalobject) in order to resolve it. So we try that first ?

Comment (sammartinez) [reply packages/amplify-ui/webpack.config.js]: @jordanranz Had this point as well https://webpack.js.org/configuration/node/ to see if we can use the polyfill for this.

Comment (powerful23) [reply packages/amplify-ui/webpack.config.js]: The polyfill is for mocking modules in NodeJS but here what we need is to not use anything specific in browser when running in the node. The `globalObject` has been changed to `this`.

**Comment by aldegoeij** [comment]
release `1.1.30-unstable.1` is still giving me this issue with `gatsby build` (SSR static html generation)

**Comment by aldegoeij** [comment]
referencing tickets #3486, #3506, #3615, #3015, #876, #3477 and webpack/webpack#6522

**Pull request merged**